include/tinyevents/tinyevents.hpp

```diff
@@ -71,20 +71,21 @@ namespace tinyevents
         }
 
         template<typename T>
-        void queue(const T &msg) {
-            queuedDispatches.push_back([msg](Dispatcher& dispatcher) {
-                dispatcher.dispatch(msg);
+        void queue(T&& msg) {
+            queuedDispatches.push([m = std::forward<T>(msg)](Dispatcher& dispatcher) {
+                dispatcher.dispatch(m);
             });
         }
 
-        void process() {
-            for (auto &queuedDispatch: queuedDispatches) {
+        inline void process() {
+            while (!queuedDispatches.empty()) {
+                std::function<void(Dispatcher&)> queuedDispatch = std::move(queuedDispatches.front());
+                queuedDispatches.pop();
                 queuedDispatch(*this);
             }
-            queuedDispatches.clear();
         }
 
-        void remove(const std::uint64_t handle) {
+        inline void remove(const std::uint64_t handle) {
```

Comment:
According to [cpp reference](https://en.cppreference.com/w/cpp/language/inline)

> A function defined entirely inside a [class/struct/union definition](https://en.cppreference.com/w/cpp/language/classes), whether it's a member function or a non-member friend function, **is implicitly an inline function** _unless it is attached to a [named module](https://en.cppreference.com/w/cpp/language/modules#Module_declarations) (since C++20)._

Do you have any specific reason for adding `inline` specifiers to these methods? I would prefer not to add specifiers that have no effect on the resulting binary code.

Reply:
You are right, will remove these again.

I had in mind that I needed them as I switch between c and c++ regularly, but as you mentioned it is implicit, so it is better to remove it again. Thanks for the hint!

Reply:
-> Removed the inline commits